enhancements/sig-architecture/224-event-based-manifestwork/README.md

```diff
@@ -0,0 +1,674 @@
+        "statusHash": "abc12345"
+        }
+    ]
+}
+```
+
+### Failure Handling
+
+~~- Retry Mechanism(TBD):~~
+
+~~In the event of a failure during event delivery or processing, a retry mechanism should be implemented to ensure reliable event propagation. When an event fails to reach its destination or encounters an error during processing, it should be retried for a certain number of times before giving up.~~
+
+~~The retry mechanism should employ exponential backoff, gradually increasing the delay between retries to prevent overwhelming the system with failed events. Each retry attempt should be logged, including relevant information such as the event payload, timestamp, and error details.~~
+
+- Resync Mechanism:
+
+To ensure the integrity of the event-based manifestworks system during source or work agent restart or reconnection, the following optimized resync mechanism is employed:
+
+1. On start, the source subscribes to the spec resync event topic /sources/resync/+/manifests from the broker.
+2. On start, the work agent subscribes to the status resync event topic /sources/+/resync/clusters/manifestsstatus from the broker.
+3. Upon work agent restart or reconnect, the work agent generates a spec resync event with resource versions and sends it to the broker.
+4. Upon source restart or reconnect, the source calculates a status hash of its owned resources and sends a status resync event with the resource hash to the broker.
+5. The source compares the resource versions in the spec resync event message with the hub's state and generates spec events to the broker for the work agent to receive missing spec events.
+6. The work agent compares the status hashes in the status resync event message with its own state and generates status events to the broker for the source to receive missing status events.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+- Unit tests
+- Integration tests
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- [Maturity levels][maturity-levels]
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, stable), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
```

Comment:
Can we add a section to describe what's changed for other components in order to use the event based framework? for example: if the policy will use this component, Policy-agent does not need to report status to hub cluster namespace. Work-agent to report the policy status. Policy-propagator controller aggregate the status based on status cloudevent. there is no need to have replicatedPolicy. we won't have cluster namespace in hub side. etc.

Reply:
sure, will add that.

Reply:
maybe add another story?

Reply:
not sure about this, but policy may not a good example for manifestwork, they have their own transport/propagation logic,  not using manifestwork.

Reply:
It is self defeating to have policy propagate on its own. We end up increasing the agent footprint on the cluster. Policy propagation has the same constraint as of manifestwork

Reply:
In fact there should be only one way in ocm to create objects in managed cluster. It should be through work agent.